### DISCUSSION TITLE
Crowd-Sourced Storytelling with Rewards

### DISCUSSION THREAD

**Discussion by SeanCheatham** [GiraffeChain/giraffe#29] [Ideas]
Giraffe Chain is meant to be mobile/web-friendly. A big benefit of putting so much web3 power into a mobile application is that protocol can be interactive.

Proof-of-stake secures the chain, but acquiring stake isn't a straightforward process. Most forms of stake acquisition require you to buy the stake with money, but Giraffe Chain isn't supposed to be just another decentralized finance instrument.

So what if stake could be acquired through other means? Something involving a bit of creativity - something that can leverage the interactive nature of a mobile phone or web browser. Something like a **crowd-sourced storytelling** platform where participants _vote_ on the next word in the story and receive rewards for their participation.

The general process I have in mind is treat it a bit like a game that takes place round-by-round. Each "round" is timed to a block in the blockchain. Adding a new word to the story takes place over the course of three rounds (three blocks).
- Round 1: Participants commit to a "word" to be added to the story. The commitment doesn't reveal the word; it is instead just a hash of the word (plus salt) to prevent plagiarism. The commitment is represented in a UTxO and must encumber some amount of tokens to prevent spam. Encumbering more tokens yields a greater reward.
- Round 2: Participants reveal the word they commited to in the previous step. This is done by spending the UTxO from Round 1 and creating a new one in its place.
- Round 3: The votes are tallied from Round 2, and the "word" with the most votes wins. To prevent an oligarchy, quadratic voting is employed. For example, encumbering a vote with 1 token equals 1 vote; encumbering a token with 4 tokens equals 2 votes; encumbering a token with 9 tokens equals 3 votes.

The process repeats indefinitely to produce a never-ending, crowd-sourced story. Participants are rewarded by spending "winning" tokens from Round 2, but at a slightly inflated value. Participants can still reclaim "losing" tokens.

While this system doesn't address the issue of bootstrapping/onboarding (i.e. starting with 0 tokens), it at least enables a user to go from 1 token to 2 tokens without spending money.

**Comment by mgrand** [comment]
It sounds like an interesting idea. The biggest problem that I see is when there is no obvious next word and so there are only words with one vote. Perhaps you can find a way for people to earn their first token by participating in a process to decide a tie.

Perhaps Giraffe chain could be used as an intellectual property market place, where the people who share the most ideas have most tokens.

**Reply by SeanCheatham** [reply]
Those are both good thoughts!

You're probably right that we'll see very little consensus on what the "next word" should be. In one sense, that may be a good thing because it limits the advantage of LLMs. On the other hand, a bunch of 1-vote words is almost entirely useless, and any resulting "story" would look like gibberish or haskell. I'm hoping that the quadratic voting mechanism might reduce the chances of a tie. If 10 people participate in a round, and they choose 10 different words, then the participant who "stakes" the most with their vote would win. While not ideal to produce another oligarchy, at least the other 9 participants _had_ a chance.

But that doesn't _prevent_ a tie, so having a tie-breaker of some sort (whether deterministic or democratic) will be necessary. Allowing 0-token holders to participate in the tie-breaker could be a really cool way to onramp people. I just don't really know how to achieve that in a sybil-resistant way right now, meaning how to prevent a person from spinning up a supercomputer which can pretend to be a million people.

The intellectual property market place is also a cool idea. I don't think LLMs can compete in this space yet which makes it really appealing. I'm wondering if you have any thoughts on how that might be accomplished? What would the UX be?
